src/wallet/wallet.cpp

```diff
@@ -1417,6 +1418,57 @@ bool CWallet::IsHDEnabled() const
     return !hdChain.masterKeyID.IsNull();
 }
 
+void CWallet::EventuallyRescanAfterKeypoolTopUp() {
+    if (fSyncPausedUntilKeypoolExt) {
+
+        // for now, enable block requests and tip updates via the states
+        // this will only be sufficient as long as only a single wallet adjusts these stats
+        // otherwise the net-/validation-logic needs to poll the state over a signal (or similar)
```

Comment:
In commit "Ensure we pause wallet sync when the keypool requires extension"

It should be possible to handle this without any polling or notifications by making `fPauseBlockRequests` and `fPauseTipUpdates` counters instead of bools. The counters would need to be incremented whenever `fSyncPausedUntilKeypoolExt` changed from false to true in any wallet, and decremented when it changed from true to false. Block requesting or tip updating would be paused whenever its respective counter was greater than 0.